core/src/main/java/io/micronaut/core/io/service/SoftServiceLoader.java

```diff
@@ -197,23 +198,20 @@ public boolean hasNext() {
                     return false;
                 }
                 URL url = serviceConfigs.nextElement();
-                try {
-                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
-                        List<String> lines = reader.lines()
-                            .filter(line -> line.length() != 0 && line.charAt(0) != '#')
-                            .filter(condition)
-                            .map(line -> {
-                                int i = line.indexOf('#');
-                                if (i > -1) {
-                                    line = line.substring(0, i);
-                                }
-                                return line;
-                            })
-                            .collect(Collectors.toList());
-                        unprocessed = lines.iterator();
-
-                    }
-                } catch (IOException e) {
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
+                    List<String> lines = reader.lines()
+                        .filter(line -> line.length() != 0 && line.charAt(0) != '#')
+                        .filter(condition)
+                        .map(line -> {
+                            int i = line.indexOf('#');
+                            if (i > -1) {
+                                line = line.substring(0, i);
+                            }
+                            return line;
+                        })
+                        .collect(Collectors.toList());
+                    unprocessed = lines.iterator();
+                } catch (IOException | UncheckedIOException e) {
```

Comment:
Not related, somehow I would get `UncheckedIOException` a lot during Gradle invocation